dts: arm: st: u0: deprecate sram0 node label

etienne-lms opened the pull request:
Deprecate sram0 node label from STM32U0xx SoC series DTSI files since it duplicates existing sram1 label that better matches SoCs reference manual wording. Update board DTSI files accordingly.

This prevents confusion for 2 nodes labels defining the same RAM but defined at different places.

Factorize definitions of `sram0` node (embedded SRAM1) and `sram2` node (embedded SRAM2) in STM32U0xx SoCs series. U031xx all have 8kB of SRAM1 and 4kB of SRAM2. U073xx and U083xx all have 32kB of SRAM1 and 8kB of SRAM2.

etienne-lms committed:
dts: arm: st: u0: deprecate sram0 node label

Deprecate sram0 node label from STM32U0xx SoC series DTSI files since it
duplicates existing sram1 label that better matches SoCs reference
manual wording. Update board DTSI files accordingly.

Still define deprecated sram0 DT node label for explicit compatibility
with out-tree boards that may rely on this legacy node label.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented on dts/arm/st/u0/stm32u031.dtsi:
```diff
@@ -7,6 +7,18 @@
+	sram0: memory@20000000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(8)>;
+		zephyr,memory-region = "SRAM1";
```
Ok, i'll restore the original name to match the node label.

etienne-lms replied:
Using better names would be orth be considered on all STM32 SoCs in a consistent way.
For now,I plan to restore the current SoCs way where `sram0` label is the default Zephyr main SRAM label, and i'll use `"SRAM0"` as region name.

JarmouniA commented on dts/arm/st/u0/stm32u073.dtsi:
```diff
@@ -72,13 +72,15 @@
 	};
 
-	sram1: memory@20000000 {
+	sram0: memory@20000000 {
```
mismatch with region name.

etienne-lms replied:
Hmm... I'd rather change my mind.

Would you (and @dsseng) be okay to rather deprecate `sram0` for these STMU0xx SoCs in favor to `sram1` label (and region naming). It would better match the SoC ref man.

Actually that's conservative since current implementation (before my changes). It would only explicitly state that `sram0` node label is deprecated. I would still define this label, for compliance with out-of-tree boards, but will be able to fully remove it after a few release tags, maybe for v4.6 LTS or next v5.x tags.

etienne-lms commented:
I've updated the change according to https://github.com/zephyrproject-rtos/zephyr/pull/98929#discussion_r2503930511.

etienne-lms commented on dts/arm/st/u0/stm32u031X4.dtsi:
```diff
@@ -7,18 +7,6 @@
-		reg = <0x20080000 DT_SIZE_K(4)>;
-		zephyr,memory-region = "SRAM2";
-	};
-
```
@erwango, I wonder if factorizing SRAMs size info in **stm32u0??.dtsi** sub-series files is a good idea. Maybe it is more convenient to have both flash and SRAM sizes all defined together in **stm32u??X?.dtsi**, for consistency with the other STM32 SoCs.
Based on feedback on this P-R related to STM32U0xx, I plan to propose equivalent changes in other SoCs to better name SRAM nodes accrdong to reference manual wording.

etienne-lms committed:
dts: arm: st: u0: use flash0 node label to set flash size

Use flash0 node label to set the internal flash size instead of
redefining the node path. This change makes SoC variants DTSI files
more consistency.

No functional change.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

erwango commented on dts/arm/st/u0/stm32u031X4.dtsi:
```diff
@@ -27,3 +15,11 @@
+
+&sram2 {
+	reg = <0x20080000 DT_SIZE_K(4)>;
+};
```
In st .dtsi files, we generally don't overwrite nodes using `&node` references but directly "in place", so it would be:
```
/ {
	sram1: memory@20000000 {
		reg = <0x20000000 DT_SIZE_K(8)>;
	};
       [...]
```
I think, this should be kept for consistency and better readability of the .dtsi nodes (as it preserves "hierarchy": for instance we see that sram, is not part of node soc/, which is not the case using &node reference.).
